### PR TITLE
dxScheduler - fix time zone converting startDate and endDate in appointment form

### PR DESCRIPTION
**Pull request by andrewmakarov**

### CODE REVIEW

File: js/ui/scheduler/appointmentPopup.js [hunk 1]
```diff
@@ -194,15 +194,15 @@ export default class AppointmentPopup {
         this.state.appointment.isEmptyText = data === undefined || data.text === undefined;
         this.state.appointment.isEmptyDescription = data === undefined || data.description === undefined;
 
-        const formData = extend({}, { text: '', description: '', recurrenceRule: '' }, this._createAppointmentFormData(data));
+        const formData = extend({ text: '', description: '', recurrenceRule: '' }, this._createAppointmentFormData(data));
 
         if(processTimeZone) {
             if(startDate) {
```
Comment (Yarovaya) [inline js/ui/scheduler/appointmentPopup.js:200]: In my opinion, the need for such verification is in doubt. Can we refactor it in this PR?

Comment (andrewmakarov) [reply js/ui/scheduler/appointmentPopup.js:200]: No problem

File: js/ui/scheduler/appointmentPopup.js [hunk 2]
```diff
@@ -194,15 +194,15 @@ export default class AppointmentPopup {
...
-                startDate = this.scheduler.fire('convertDateByTimezone', startDate);
+                startDate = this.scheduler.fire('convertDateByTimezone', startDate, this.scheduler.fire('getField', 'startDateTimeZone', data), true);
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/appointmentPopup.js]: To my mind `this.scheduler.fire('getField', 'startDateTimeZone', data)` should be in a separate variable `startDateTimeZone`

File: js/ui/scheduler/appointmentPopup.js [hunk 3]
```diff
@@ -194,15 +194,15 @@ export default class AppointmentPopup {
...
                 this.scheduler.fire('setField', 'startDate', formData, startDate);
             }
             if(endDate) {
-                endDate = this.scheduler.fire('convertDateByTimezone', endDate);
+                endDate = this.scheduler.fire('convertDateByTimezone', endDate, this.scheduler.fire('getField', 'endDateTimeZone', data), true);
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/appointmentPopup.js]: And here..

**Review by Yarovaya — APPROVED** [review]

**Review by AntonSermyazhko — APPROVED** [review]

**Review by AryamnovEugeniy — APPROVED** [review]

**Pull request merged**